src/Formatters/VarDumpFormatter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Consolidation\OutputFormatters\Formatters;
+
+use Consolidation\OutputFormatters\Options\FormatterOptions;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+
+/**
+ * Var_dump formatter
+ *
+ * Run provided data through Symfony VarDumper component.
+ */
+class VarDumpFormatter implements FormatterInterface
+{
+    /**
+     * @inheritdoc
+     */
+    public function write(OutputInterface $output, $data, FormatterOptions $options)
+    {
+        /** @var \Symfony\Component\Console\Output\StreamOutput $output */
+        (new CliDumper())->dump(
+            (new VarCloner())->cloneVar($data),
+            $output->getStream()
```

Comment:
Didn't notice this until you wrote the tests, but `getStream` is not a method of OutputInterface, so this usage violates the contract of the `write` method.

It would be all right to use `$output->getStream()` as an optimization, if $output isA StreamOutput. However, if it is not, then the CliDumper should write to a temporary stream, and the result should then be written to $output, so that any OutputInterface may be used.

Reply:
Ah, I see your note above on OutputInterface support. If the output isa StreamOutput, then use the color-aware `getStream` method; otherwise, use writeln as you show above.